test/TesterAzureUtils/Streaming/StreamReliabilityTests.cs

```diff
@@ -47,7 +47,6 @@ public override TestCluster CreateTestCluster()
 
             this.numExpectedSilos = 2;
             var options = new TestClusterOptions(initialSilosCount: (short)this.numExpectedSilos);
-            options.ClusterConfiguration.Globals.LivenessType = GlobalConfiguration.LivenessProviderType.AzureTable;
```

Comment:
tests are failing due to this. Since you are removing the LivenessType setting from the GlobalConfiguration, then when the ClientConfiguration is created in the TestClusterOptions.BuildClientConfiguration() method,  the client is not configured to use GatewayProviderType.AzureTable setting. You'll have to set this setting yourself (do it after ClusterConfiguration has been fully configured, because order matters):
```
options.ClientConfiguration.GatewayProvider = ClientConfiguration.GatewayProviderType.AzureTable;
```

Please do this on all tests where you moved off the `Globals.LivenessType` to set the membership provider

Reply:
Why is liveness provider type been copied to client config side? What's the reason for TestCluster doing this behavior?

Reply:
Convenience. So that if you configure a non-default liveness type on the test cluster, it is automatically converted to it's par on the client side.
I understand we loose that ability once we switch to extension methods, but to be fair this is mostly a convenience for Orleans' internal testing, as most of the times end users will not change the liveness for their in-memory tests...

Reply:
I'm just saying this is pretty confusing, and probably confusing for end users too. I didn't know why the decision was made that way since I wasn't there. So I was just asking for the reason then.

Reply:
yes, agree it's confusing. Nevertheless it's pretty low impact, since most of our users just use the `TestCluster` but don't change the membership. The only reason to change the membership is when you are actually testing a custom membership or restarting all silos (and hence the GrainBased one miss-behaves), which is *mostly* just us, and a few rare exceptions.
Anyway, once we move away from the legacy configuration also in `TestCluster,` this *convenience transformation* will likely go away too